### DISCUSSION TITLE
How to factorize examples?

### DISCUSSION THREAD

**Discussion by bfontaine** [pydantic/pydantic#9172] [Question] [answered]
Hello,

I have some types I reuse in multiple models, but I need to repeat the `examples` every time:

```python
MyID = Annotated[str, StringConstraints(pattern=r"^ABC\d{3}$")]

class Foo(BaseModel):
    id: MyID = Field(examples=["ABC123"])

class Bar(BaseModel):
    foo_id: MyID = Field(examples=["ABC123"])
```

I can’t find a way to move the examples on the annotated type instead, although it would make sense because these exemples are inherent to the type itself and not where the fields are defined.

Is there some way to factorize this, so I can use `id: MyID` without re-defining a `Field()` every time? I could store the `Field()` in a variable and use `<field>: MyID = my_id_field()` but that’s still verbose.

Thanks in advance.

**Comment by uriyyo** [comment, marked as answer]
It's should be possible to use field on annotated level:
```py
MyID = Annotated[
    str,
    StringConstraints(pattern=r"^ABC\d{3}$"),
    Field(examples=["ABC123"]),
]


class Foo(BaseModel):
    id: MyID


class Bar(BaseModel):
    foo_id: MyID
```

**Reply by bfontaine** [reply]
Thanks!
